starknet_devnet/starknet_wrapper.py

```diff
@@ -348,12 +376,15 @@ async def __aexit__(
                     transaction_hash=tx_hash,
                 )
 
-                await self.starknet_wrapper._store_transaction(
-                    transaction=transaction,
-                    state_update=state_update,
-                    error_message=error_message,
-                    tx_hash=tx_hash,
-                )
+                if self.starknet_wrapper.config.blocks_on_demand:
+                    self.starknet_wrapper.mempool.append(transaction)
```

Comment:
When and where is mempool cleared?

Reply:
Good that you spotted that bug, it should be cleared in `store_mempool_transactions()` but it's not... I can't explain why I forgot about this. I'll also add this to tests.

Reply:
Fixed in 7258b975bc3d67cd3b6ef21968beb844c927d03c